File: regexlib/2021-5-15/javascript_test_file/regexlib_4131.js
/* 4131
 * ^("(\\["\\]|[^"])*"|[^\n])*(\n|$)
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"""+"\\"*32+"! _1_EOA(iv)"
 */
var REGEX = /^("(\\["\\]|[^"])*"|[^\n])*(\n|$)/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '\"' + '\\'.repeat(i*1) + '! _1_EOA(iv)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_4131.py
# 4131
# ^("(\\["\\]|[^"])*"|[^\n])*(\n|$)
# EXPONENT
# nums:4
# EXPONENT AttackString:"""+"\\"*32+"! _1_EOA(iv)"

import re
from time import perf_counter

regex = """^("(\\["\\]|[^"])*"|[^\n])*(\n|$)"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "\"" + "\\" * i * 1 + "! _1_EOA(iv)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")